contracts/staking/PoSValidatorManager.sol

```diff
@@ -92,34 +92,38 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         uint32 messageIndex
     ) external {
         if (includeUptimeProof) {
-            PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
-            (WarpMessage memory warpMessage, bool valid) =
-                WARP_MESSENGER.getVerifiedWarpMessage(messageIndex);
-            require(valid, "PoSValidatorManager: invalid warp message");
-
-            require(
-                warpMessage.sourceChainID == WARP_MESSENGER.getBlockchainID(),
-                "PoSValidatorManager: invalid source chain ID"
-            );
-            require(
-                warpMessage.originSenderAddress == address(0),
-                "PoSValidatorManager: invalid origin sender address"
-            );
-
-            (bytes32 uptimeValidationID, uint64 uptime) =
-                ValidatorMessages.unpackValidationUptimeMessage(warpMessage.payload);
-            require(
-                validationID == uptimeValidationID,
-                "PoSValidatorManager: invalid uptime validation ID"
-            );
-
-            $._validatorUptimes[validationID] = uptime;
-            emit ValidationUptimeUpdated(validationID, uptime);
+            _updateUptime(validationID, messageIndex);
         }
-
         _initializeEndValidation(validationID);
     }
 
+    function _updateUptime(bytes32 validationID, uint32 messageIndex) internal returns (uint64) {
+        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+        (WarpMessage memory warpMessage, bool valid) =
+            WARP_MESSENGER.getVerifiedWarpMessage(messageIndex);
+        require(valid, "PoSValidatorManager: invalid warp message");
+
+        require(
+            warpMessage.sourceChainID == WARP_MESSENGER.getBlockchainID(),
+            "PoSValidatorManager: invalid source chain ID"
+        );
+        require(
+            warpMessage.originSenderAddress == address(0),
+            "PoSValidatorManager: invalid origin sender address"
+        );
+
+        (bytes32 uptimeValidationID, uint64 uptime) =
+            ValidatorMessages.unpackValidationUptimeMessage(warpMessage.payload);
+        require(
+            validationID == uptimeValidationID, "PoSValidatorManager: invalid uptime validation ID"
+        );
+
+        $._validatorUptimes[validationID] = uptime;
+        emit ValidationUptimeUpdated(validationID, uptime);
```

Comment:
We should only update the validator's uptime if the uptime provided is greater than the previously provided uptime. This way delegators may not have to provide an uptime proof to get their rewards, provided the validator has had a sufficiently high uptime.